### DISCUSSION TITLE
Template Functions

### DISCUSSION THREAD

**Discussion by khrome83** [pocketbase/pocketbase#3431] [Q&A] [answered]
Is there a mechanism to attach new functions to the `html/template` interface Pocketbase uses when extended with Go?

I can see how we can load files, string, or the FS.

How can we append functions? For example i am looking to make a modulus function. 

For now, I plan to move this logic into the handler. But it would be clearer in the template, as the template has the final shaped object and count. It will be harder for me to get a proper count in the handler.

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
Not directly at the moment (_I've added it to the roadmap to consider it for the next release_).

As a workaround you can define a dummy util/helper struct and attach methods to it. For example:

```go
type Utils struct {
}

func (u Utils) ToUpper(a string) string {
	return strings.ToUpper(a)
}

...

data := map[string]any{"name": "John", "utils": Utils{}}

html, err := template.NewRegistry().LoadFiles(
	"views/base.html",
	"views/hello.html",
).Render(data)
```

And from the template it can be called with something like:
```
Hello {{.utils.ToUpper "something"}}
```

**Reply by khrome83** [reply]
Thank you!
